.github/workflows/pr_build.yaml

```diff
@@ -27,12 +28,70 @@ jobs:
           go-version: ${{ env.GO_VERSION }}
       - name: Test
         run: make test
+
+  lint-windows:
+    runs-on: windows-2022
+    defaults:
+      run:
+        shell: msys2 {0}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Setup go
+        uses: actions/setup-go@v2
+        with:
+          go-version: ${{ env.GO_VERSION }}
+      - name: Install msys2
+        uses: msys2/setup-msys2@v2
+        with:
+          msystem: MINGW64
+          update: true
+          install: >-
+            git
+            base-devel 
+            mingw-w64-x86_64-toolchain 
+            unzip 
+      - name: Lint
+        run: make lint
+
+  test-windows:
+    runs-on: windows-2022
+    defaults:
+      run:
+        shell: msys2 {0}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Setup go
+        uses: actions/setup-go@v2
+        with:
+          go-version: ${{ env.GO_VERSION }}
+      - name: Install msys2
+        uses: msys2/setup-msys2@v2
+        with:
+          msystem: MINGW64
+          update: true
+          install: >-
+            git
+            base-devel 
+            mingw-w64-x86_64-toolchain 
+            unzip 
+      - name: Test
+        run: make test
+
   # This job is just here to make sure that the other jobs have completed
   # and is used as a single job to block PR merge from. GH doesn't have a
   # way to say "all jobs from this action", which would be ideal.
-  success:
-    needs: [lint, test]
+  success-linux:
```

Comment:
may we put all need in the same job? if we separate between windows/linux, we'll not have a single job that group all.
```
success:
    needs: [lint-linux, test-linux, lint-windows, test-windows]
    runs-on: ubuntu-latest
    steps:
      - name: Shout it out
        run: echo SUCCESS
      - name: Shout it out - Linux
        run: echo SUCCESS - Linux
```